internal/models/session.go

```diff
@@ -0,0 +1,133 @@
+package models
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/ydataai/go-core/pkg/common/logging"
+)
+
+// SessionStorage is a struct to temporarily saving the session.
+type SessionStorage struct {
+	configuration SessionStorageConfiguration
+	sessions      map[string]*Session
+	mtx           sync.RWMutex
+	logger        logging.Logger
+}
+
+// NewSessionStorage creates a new temporary session and saves for validation.
+func NewSessionStorage(logger logging.Logger, config SessionStorageConfiguration) *SessionStorage {
+	ss := &SessionStorage{
+		configuration: config,
+		logger:        logger,
+		sessions:      make(map[string]*Session),
```

Comment:
why are you storing a pointer?

Reply:
because the Service was a pointer, but I solved it.